### PR TITLE
Feature/asv 172 my account settings fragment

### PR DESCRIPTION
**Pull request by paquiquinho**
**What does this PR do?**

This PR aims to change the account view and link settings from there. It also changed most text related to the settings themselves.

**Database changed?**

   No

**Where should the reviewer start?**

- [x] NewAccountFragment.java
- [x] NewAccountPresenter.java
- [x] NewAccountNavigator.java
- [x] new_account_layout.xml

**How should this be manually tested?**

Open from the apps icon in the bottom navigation at this point. Every flow can be tested there. Flows can be found [here](https://zpl.io/aBOX9xA).

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-172](<https://aptoide.atlassian.net/browse/ASV-172>)

**Code Review Checklist**

- [x] Architecture
- [x] Documentation on public interfaces
- [x] Database changed?
- [x] If yes - Database migration?
- [x] Remove comments & unused code & forgotten testing Logs
- [x] Codestyle
- [x] Partners build
- [x] Unit tests pass
- [x] Functional QA tests pass

### CODE REVIEW

File: app/src/main/res/values/strings_accountmanager.xml [hunk 1]
```diff
@@ -95,4 +95,15 @@
...
+  <!-- New Account Strings -->
+  <string name="newaccount_signin_message">Sign in or register to get the most out of Aptoide</string>
+  <string name="newaccount_signin">LOGIN/REGISTER</string>
+  <string name="newaccount_my_account">My account</string>
+  <string name="newaccount_my_store">My store</string>
+  <string name="newaccount_create_store">CREATE A STORE</string>
+  <string name="newaccount_create_store_message">Create an Aptoide store to share apps with the community.</string>
+  <string name="newaccount_find_friends">FIND FRIENDS</string>
+  <string name="newaccount_log_out">LOGOUT</string>
+  <string name="newaccount_notification_history">Navigation history</string>
```
Comment (jdandrade) [inline app/src/main/res/values/strings_accountmanager.xml]: "Navigation history"

File: app/src/main/res/layout/new_account_layout.xml [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        android:orientation="horizontal"
+        >
+      <ImageView
+          android:layout_width="24dp"
+          android:layout_height="24dp"
+          android:layout_marginBottom="16dp"
+          android:layout_marginLeft="16dp"
+          android:layout_marginStart="16dp"
+          android:layout_marginTop="16dp"
+          android:src="@drawable/ic_action_comments_light"
```
Comment (jdandrade) [inline app/src/main/res/layout/new_account_layout.xml]: settings icon

File: app/src/main/res/layout/new_account_layout.xml [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
```
Comment (jdandrade) [inline app/src/main/res/layout/new_account_layout.xml]: notification bell icon

File: app/src/main/res/values/strings_settings.xml [hunk 4]
```diff
@@ -5,58 +5,50 @@
...
-  <string name="setting_mature_filtering">Adult Content</string>
-  <string name="setting_dont_show_mature_content">Show mature content</string>
-  <string name="set_mature_pin_title">Set adult content pin</string>
-  <string name="set_mature_pin_summary">Set a pin that will be required to unlock adult content</string>
-  <string name="remove_mature_pin_title">Remove adult content pin</string>
-  <string name="remove_mature_pin_summary">Remove the pin required to unlock adult content</string>
-
+  <string name="setting_appfiltertitle">Only show compatible apps</string>
+  <string name="setting_appfiltersum">You\'ll only see apps compatible with this device</string>
+  <string name="setting_mobile">Download only over wifi (WRONG IMPLEMENTATION)</string>
```
Comment (jdandrade) [inline app/src/main/res/values/strings_settings.xml]: "wrong implementation"

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(account -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void handleProfileEditClick() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(viewCreated -> view.editUserProfileClick()
+            .flatMap(click -> accountManager.accountStatus())
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: accountStatus().first()

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/SettingsFragment.java [hunk 6]
```diff
@@ -229,9 +242,9 @@ private boolean shouldRefreshUpdates(String key) {
 
   private void setupClickHandlers() {
     //set AppStore name
-    findPreference(SettingsConstants.CHECK_AUTO_UPDATE_CATEGORY).setTitle(
-        AptoideUtils.StringU.getFormattedString(R.string.setting_category_autoupdate,
-            getContext().getResources(), marketName));
+    //findPreference(SettingsConstants.CHECK_AUTO_UPDATE_CATEGORY).setTitle(
+    //    AptoideUtils.StringU.getFormattedString(R.string.setting_category_autoupdate,
+    //        getContext().getResources(), marketName));
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/view/settings/SettingsFragment.java]: commented code

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+  }
+
+  @Override public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String s) {
+
+  }
+
+  @Override public void showAccount(Account account) {
+    if (account.getNickname()
+        .isEmpty() || account.getNickname()
+        .equals(null) || account == null) {
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java]: if account is null this throws an exception so no need account == null

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java (hunk 5 above)
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/SettingsFragment.java (hunk 6 above)
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/SettingsFragment.java]: fixed

File: app/src/main/res/layout/new_account_layout.xml [hunk 8]
```diff
@@ -0,0 +1,202 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/new_account_layout.xml]: fixed

File: app/src/main/res/layout/store_profile_account_displayable.xml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        android:layout_height="wrap_content"
+        android:text="My Profile/Store"
+        android:textColor="@color/black"
+        android:textStyle="bold"
+        />
+    <TextView
+        android:id="@+id/description"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Batatas"
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/store_profile_account_displayable.xml]: fixed

File: app/src/main/res/layout/store_profile_account_displayable.xml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+      android:layout_marginStart="10dp"
+      android:layout_toEndOf="@id/user_icon"
+      android:layout_toRightOf="@id/user_icon"
+      android:orientation="vertical"
+      >
+    <TextView
+        android:id="@+id/name"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="My Profile/Store"
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/store_profile_account_displayable.xml]: fixed

File: app/src/main/res/layout/new_account_layout.xml [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+      <TextView
+          android:id="@+id/create_store_message"
+          android:layout_width="250dp"
+          android:layout_height="wrap_content"
+          android:layout_gravity="center_horizontal"
+          android:layout_marginTop="16dp"
+          android:gravity="center"
+          android:text="@string/newaccount_create_store_message"
+          android:textColor="@color/black"
+          android:textSize="17dp"
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/new_account_layout.xml]: fixed

File: app/src/main/res/layout/new_account_layout.xml [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+          android:id="@+id/logout_button"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:layout_gravity="center_horizontal"
+          android:layout_marginEnd="16dp"
+          android:layout_marginRight="16dp"
+          android:layout_marginStart="16dp"
+          android:layout_marginTop="8dp"
+          android:background="@color/aptoide_orange"
+          android:text="@string/newaccount_log_out"
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/new_account_layout.xml]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+  private Button findFriendsButton;
+  private Button createStoreButton;
+  private Button editStoreButton;
+  private Button editProfileButton;
+
+  //Navigation buttons
+  private View notificationHistory;
+  private View settings;
+
+  //TODO: Add string resources to the settings XML and fragment (ALL STRINGS HARDCODED!!!)
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java (hunk 7 above)
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+    this.view = view;
+    this.accountManager = accountManager;
+    this.crashReport = crashReport;
+    this.sharedPreferences = sharedPreferences;
+    this.scheduler = scheduler;
+    this.newAccountNavigator = newAccountNavigator;
+  }
+
+  @Override public void present() {
+    //Account handlers
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: comment

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+import rx.Observable;
+import rx.Scheduler;
+
+/**
+ * Created by franciscocalado on 13/03/18.
+ */
+
+public class NewAccountPresenter implements Presenter {
+
+  public static final int EDIT_STORE_REQUEST_CODE = 1230;
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: Can be private if there is no necessity of exposing it to outer classes

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+        .putBoolean(ManagedKeys.TWITTER_SYNC, false)
+        .putBoolean(ManagedKeys.FACEBOOK_SYNC, false)
+        .apply();
+  }
+
+  private boolean storeExistsInAccount(Account account) {
+    return account.getStore()
+        .getId() != 0;
+  }
+}
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: this presenter is missing some observeOn's on the main thread

File: aptoidepreferences/src/main/java/cm/aptoide/pt/preferences/managed/ManagerPreferences.java [hunk 17]
```diff
@@ -56,7 +56,7 @@ public static boolean getGeneralDownloadsWifi(SharedPreferences sharedPreference
   }
 
   public static boolean getGeneralDownloadsMobile(SharedPreferences sharedPreferences) {
-    return sharedPreferences.getBoolean(ManagedKeys.GENERAL_DOWNLOADS_MOBILE, true);
+    return !sharedPreferences.getBoolean(ManagedKeys.GENERAL_DOWNLOADS_MOBILE, true);
```
Comment (jdandrade) [inline aptoidepreferences/src/main/java/cm/aptoide/pt/preferences/managed/ManagerPreferences.java]: in this specific situation, should we comment the code? would it help to the next dev to understand why is this returning the reverse? would it even matter?

File: app/src/main/res/layout/store_profile_account_displayable.xml [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+  <Button
+      android:id="@+id/edit_button"
+      android:layout_width="70dp"
+      android:layout_height="40dp"
+      android:layout_alignParentEnd="true"
+      android:layout_alignParentRight="true"
+      android:layout_centerVertical="true"
+      android:layout_marginEnd="10dp"
+      android:layout_marginRight="10dp"
+      android:text="Edit"
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/store_profile_account_displayable.xml]: fixed

File: app/src/main/res/layout/new_account_layout.xml [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+          android:id="@+id/find_friends_button"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:layout_gravity="center_horizontal"
+          android:layout_marginEnd="16dp"
+          android:layout_marginRight="16dp"
+          android:layout_marginStart="16dp"
+          android:layout_marginTop="8dp"
+          android:background="@color/aptoide_orange"
+          android:text="@string/newaccount_find_friends"
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/res/layout/new_account_layout.xml]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java (hunk 14 above)
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java (hunk 15 above)
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: fixed

File: app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java (hunk 16 above)
Comment (paquiquinho) [reply app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java]: fixed

**Pull request merged**